src/Sarif/Readers/ResultDiffingVisitor.cs

```diff
@@ -31,7 +31,7 @@ public bool Diff(IEnumerable<Result> actual)
 
             this.AbsentResults = this.SharedResults;
 
-            this.SharedResults = new HashSet<Result>();
+            this.SharedResults = new HashSet<Result>(ResultEqualityComparer.Instance);
```

Comment:
... so I suggest the change for two reasons:

1. Tests should use the public API as far as possible, so they test what customers could do.
2. The public API is shorter and more readable, and does not introduce additional classes (the class `ResultEqualityComparer` and all the concrete comparer classes are internal.